### PR TITLE
 Change Blog Search Engine Visibility Notice to AIOSEOP_Notices class

### PR DESCRIPTION
**Pull request by EkoJR**
Issue #1854

### CODE REVIEW

**Comment by arnaudbroes** [comment]
This is good, but again the wording might need some tweaking. I'd also remove the button and link to options-reading.php in the text. 

More importantly, I'd make sure people know this notice is being displayed by All in One SEO Pack so they're aware we are providing them with this valuable insight.

image

Here's what I suggest:

Warning: All in One SEO Pack has detected that you're blocking access to search engines. You can change this in <a href="/wp-admin/">Reading Settings</a> if this is unintended.

Remind me later.

@michaeltorbert @wpsmort what do you think?

**Comment by wpsmort** [comment]
@arnaudbroes Looks good, the only changes I'd make are:

- Change "you're" to "you are".  I believe contractions make translation for some languages more difficult (although I may be wrong about this)
- Change "Reading" to "Settings > Reading" which hopefully makes it clearer where they should navigate to

**Comment by EkoJR** [comment]
@arnaudbroes @wpsmort Regarding "Settings > Reading", I noticed that is for WP versions under 3.5, and we're long past that. So we can just use what WP currently calls it, which is "Reading Settings"

**Comment by arnaudbroes** [comment]
image
@EkoJR link is no longer clickable.

**Comment by wpsmort** [comment]
@EkoJR My suggestion of Settings > Reading was in reference to how you reach that screen but going to Settings and then Reading.

**Comment by EkoJR** [comment]
I believe I mentioned a limitations with notice messages, and one of them is the message itself can't have links, which is part of the reason why I made action_options to act as links as well.

So, I had to add the link back to the action_option where "Remind me later" is located, and it's changed to "Settings > Reading"

**Comment by arnaudbroes** [comment]
@EkoJR I wasn't aware notices can't contain links. Can you quickly change those action links to buttons (as I think that will look a bit better)? Then this should be good to go.

File: admin/functions-notice.php [hunk 1]
```diff
@@ -85,4 +85,84 @@ function aioseop_notice_disable_woocommerce_detected_on_nonpro() {
...
 		$aioseop_notices->deactivate_notice( 'woocommerce_detected' );
 	}
+
+	/**
+	 * Set Notice for Disabled Public Blog
+	 *
+	 * Admin Notice when "Discourage search engines from indexing this site" is
+	 * enabled in Settings > Reading.
+	 *
+	 * @since 2.4.2
```
Comment (contactashish13) [inline admin/functions-notice.php]: shouldn't this (and others below) be since 3.0?

File: admin/functions-notice.php [hunk 2]
```diff
@@ -85,4 +85,84 @@ function aioseop_notice_disable_woocommerce_detected_on_nonpro() {
...
+	}
+
+	/**
+	 * Notice - Blog Visibility
+	 *
+	 * Displays when blog disables search engines from indexing.
+	 *
+	 * @since 2.4.2
+	 *
+	 * @return array
```
Comment (contactashish13) [inline admin/functions-notice.php]: a description of what it returns would help

File: admin/functions-notice.php (hunk 1 above)
Comment (EkoJR) [reply admin/functions-notice.php]: Yea it should. Changed

File: admin/functions-notice.php (hunk 2 above)
Comment (EkoJR) [reply admin/functions-notice.php]: Added

**Pull request closed without merging**